tests/ref/root_display_without_style_a.html

```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script>
+  document.removeChild(document.documentElement);
+  root = document.createElementNS(null, "html");
+  body = document.createElement("body");
+  body.appendChild(document.createTextNode("Hello world!"));
+  root.appendChild(body);
+  document.appendChild(root);
```

Comment:
For elements created with namespace `null`, the user agent stylesheet is not processed. This means that the root element created in this test has no style information attached with it, so that `longhands::display::cascade_property_custom` is never called. This used to be the only place where `longhands::_servo_display_for_hypothetical_box::derive_from_display` was being called, so that my change to this function alone would not pass this test. This is why I have added a call to this function in the `!seen.get_display() ` block, making sure that `_servo_display_for_hypothetical_box` also gets the correct value when there is no display style information for the root element.

(I have added a body element in the test above to get its margin from the user style sheet. Otherwise I would have needed a second reference file with `<body style="margin:0">`.)